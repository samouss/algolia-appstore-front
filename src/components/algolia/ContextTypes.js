import PropTypes from 'prop-types';

export default {
  algoliaClient: PropTypes.object.isRequired,
  algoliaHelper: PropTypes.object.isRequired,
};
