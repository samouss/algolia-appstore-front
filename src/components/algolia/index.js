export { default as Provider } from './Provider';
export { default as withPaginateResults } from './withPaginateResults';
