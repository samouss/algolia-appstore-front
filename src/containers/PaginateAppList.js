import { withPaginateResults } from 'components/Algolia';
import AppList from 'components/AppList';

const PaginateAppList = withPaginateResults({
  hitsPerPage: 25,
})(AppList);

export default PaginateAppList;
