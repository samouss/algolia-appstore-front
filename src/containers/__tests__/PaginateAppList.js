import React from 'react';
import { shallow } from 'enzyme';
import { createMockAlgoliaClient, createMockAlgoliaHelper } from 'test/algolia';
import PaginateAppList from '../PaginateAppList';

describe('<PaginateAppList />', () => {
  it('expect to render', () => {
    const context = {
      algoliaClient: createMockAlgoliaClient(),
      algoliaHelper: createMockAlgoliaHelper(),
    };

    const component = shallow(
      <PaginateAppList />,
      { context },
    );

    expect(component).toMatchSnapshot();
  });
});
