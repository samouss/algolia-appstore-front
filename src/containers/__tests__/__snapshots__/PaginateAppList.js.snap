// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PaginateAppList /> expect to render 1`] = `
<AppList
  hits={Array []}
  isEndReached={false}
  isInitialLoad={true}
  isLoading={true}
  nbHits={0}
  nbPages={0}
  onNextPage={[Function]}
  page={1}
  processingTimeMS={0}
/>
`;
